joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForExpressionsCreator.scala

```diff
@@ -140,4 +140,12 @@ trait AstForExpressionsCreator { this: AstCreator =>
     whileAst(testAst, bodyAst, Some(ctx.getText), line(ctx), column(ctx))
   }
 
+  protected def astForForExpression(ctx: ForExpressionContext): Ast = {
+    val forVarAst  = astForForVariableContext(ctx.forVariable())
+    val forExprAst = astForExpressionOrCommand(ctx.expressionOrCommand())
+    val forBodyAst = astForCompoundStatement(ctx.doClause().compoundStatement())
+    // TODO: for X in Y is not properly modelled by while Y
+    whileAst(forExprAst.headOption, forBodyAst, Some(ctx.getText), line(ctx), column(ctx)).withChild(forVarAst.head)
```

Comment:
since you're doing a refactoring, I suggest you refrain from using `.head` and instead have something like:
```
val root = whileAst(...
forVarAst.headOption
  .map(root.withChild(_))
  .getOrElse(root)
```